tools/amd_build/patches/a_aten_src_ATen_native_cuda_Distributions.cu.patch

```diff
@@ -0,0 +1,36 @@
+diff --git a/aten/src/ATen/native/cuda/Distributions.cu b/aten/src/ATen/native/cuda/Distributions.cu
+index 50d162f14..11b638d12 100644
+--- a/aten/src/ATen/native/cuda/Distributions.cu
++++ b/aten/src/ATen/native/cuda/Distributions.cu
+@@ -1,4 +1,4 @@
+-#include "ATen/Dispatch.h"
++
+ #include "ATen/NativeFunctions.h"
+ #include "ATen/cuda/CUDAApplyUtils.cuh"
+ 
+@@ -15,6 +15,7 @@
+ 
+ #include <cstdint>
+ #include <utility>
++#if !defined(__HIP_PLATFORM_HCC__)
+ 
+ THCGenerator* THCRandom_getGenerator(THCState* state);
+ 
+@@ -49,11 +50,16 @@ void poisson_cuda_kernel(
+ namespace at { namespace native {
+ Tensor _s_poisson_cuda(const Tensor& lambda, Generator* gen) {
+   Tensor ret = lambda.type().tensor(lambda.sizes());
++
++  #if !defined(__HIP_PLATFORM_HCC__)
+   auto lambda_ = lambda.toType(ScalarType::Float);
+   AT_DISPATCH_FLOATING_TYPES(ret.type(), "poisson", [&] {
+      poisson_cuda_kernel<scalar_t>(ret, lambda_, next_philox_seed(gen));
+    });
++  #endif
```

Comment:
This looks really dangerous, because it will basically silently return an uninitialized tensor if you compile with HIP